### PR TITLE
Practice data

### PR DESCRIPTION
**Pull request by TomWoodward**
for: openstax/unified#1363

### CODE REVIEW

**Comment by TomWoodward** [comment]
ah i forgot to generate the summary, will add that

File: concourse/import-exercises/script.deno.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+const Authorization = assertDefined(Deno.env.get('EXERCISES_AUTHORIZATION'), 'EXERCISES_AUTHORIZATION environment variable must be set');
+const csvPath = assertDefined(Deno.env.get('CSV_FILE'), 'CSV_FILE environment variable must be set');
+
+// only physics has practice right now, probably add this to the csv in the future
+const bookId = 'cce64fde-f448-43b8-ae88-27705cceb0da';
+
+const f = await Deno.open(csvPath);
+
+const pages: {[key: string]: any[]} = {};
```
Comment (PiotrKozlowski) [inline concourse/import-exercises/script.deno.ts:30]: Can we create an interface for this, for `row of readCSVObjects(f)` and for `exercise`?

File: concourse/import-exercises/script.deno.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    .then((response) => response.json())
+  ;
+
+  (pages[row.source_page] = pages[row.source_page] || []).push({
+    answers: exercise.questions[0].answers,
+    group_uuid: exercise.group_uuid,
+    stem_html: exercise.questions[0].stem_html,
+    tags: exercise.group_uuid,
+    uid: exercise.uid,
+  });
```
Comment (PiotrKozlowski) [inline concourse/import-exercises/script.deno.ts:49]: Currently we are using this interface:
```ts
interface PracticeQuestion {
  uuid: string;
  stem_html: string;
  id: number;
  answers: PracticeAnswer[];
}
```
Either this interface or this property names should be adjusted

File: concourse/import-exercises/script.deno.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
+import { readCSVObjects } from 'https://deno.land/x/csv/mod.ts';
```
Comment (PiotrKozlowski) [inline concourse/import-exercises/script.deno.ts:1]: I have an extension in vscode that is showing me a bunch of lint errors in this file according to `tslint.json`
I guess it doesn't really matter if this works

**Review by PiotrKozlowski — CHANGES_REQUESTED** [review]

File: concourse/import-exercises/script.deno.ts (hunk 2 above)
Comment (TomWoodward) [reply concourse/import-exercises/script.deno.ts]: i'll update the type, its intentionally changed

File: concourse/import-exercises/script.deno.ts (hunk 1 above)
Comment (TomWoodward) [reply concourse/import-exercises/script.deno.ts]: the future of this script is uncertain so i'm not going to try super hard to make the types correct

File: concourse/import-exercises/script.deno.ts (hunk 3 above)
Comment (TomWoodward) [reply concourse/import-exercises/script.deno.ts:1]: yea i haven't found a linter that works for deno very well yet, the import format and the top level await anger my tslint

**Comment by TomWoodward** [comment]
@PiotrKozlowski i added the summary data file and had to reorganize a lot of stuff to get mocking working correctly in the pq gateway, please review again

**Review by PiotrKozlowski — APPROVED** [review]

**Pull request merged**